app/models/training_period.rb

```diff
@@ -12,13 +12,17 @@ class TrainingPeriod < ApplicationRecord
   belongs_to :ect_at_school_period, class_name: "ECTAtSchoolPeriod", inverse_of: :training_periods
   belongs_to :mentor_at_school_period, inverse_of: :training_periods
   belongs_to :school_partnership
-  belongs_to :expression_of_interest, class_name: 'ActiveLeadProvider'
+
   has_one :lead_provider_delivery_partnership, through: :school_partnership
   has_one :active_lead_provider, through: :lead_provider_delivery_partnership
   has_one :lead_provider, through: :active_lead_provider
   has_one :delivery_partner, through: :lead_provider_delivery_partnership
   has_one :contract_period, through: :active_lead_provider
 
+  belongs_to :expression_of_interest, class_name: 'ActiveLeadProvider'
+  has_one :expression_of_interest_lead_provider, through: :expression_of_interest, class_name: 'LeadProvider', source: :lead_provider
+  has_one :expression_of_interest_contract_period, through: :expression_of_interest, class_name: 'LeadProvider', source: :contract_period
```

Comment:
Typo? 😅

```suggestion
  has_one :expression_of_interest_contract_period, through: :expression_of_interest, class_name: 'ContractPeriod', source: :contract_period
```

Can the `class_name` just be dropped entirely? [I _think_ rails infers it from the source](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_one).

<img width="977" height="86" alt="image" src="https://github.com/user-attachments/assets/373c2ad4-58ec-4585-b4c9-5b2193bf87d5" />